python/tests/test_sklearn_vw.py

```diff
@@ -184,6 +186,20 @@ def test_decision_function(self, data):
 
         assert np.allclose(expected, actual)
 
+    def test_shuffle_pd_Series(self):
+        # dummy data in vw format
+        X = pd.Series(['1 |Pet cat', '-1 |Pet dog', '1 |Pet cat', '1 |Pet cat'], name='catdog')
+
+        kfold = KFold(n_splits=3, random_state=314, shuffle=False)
+        for train_idx, valid_idx in kfold.split(X):
+            X_train = X[train_idx]
+            # Classifier with multiple passes over the data
+            clf = VWClassifier(passes=3, convert_to_vw=False)
+            # Test that there is no exception raised in the fit on folds
+            try:
+                clf.fit(X_train)
+            except KeyError:
```

Comment:
catching exceptions is not really necessary under pytest, if an exception occurs it will be caught and the test marked as a failure

Reply:
OK, but the intention was to clarify with the error message what is the cause of the problem (as otherwise the exception itself is hard to trace). Did I fail to make an instructive message?

Reply:
it's fine, it's just a level of rigor in the test harness that I've never seen before =)
under pytest it's easy to trace the problem back to this test, but we can keep it here.